File: scratch/test_regex.py
import json

text = "#32=CARTESIAN_POINT('',(46.765612334,29.,5.));"
import re
regex = re.compile(r"CARTESIAN_POINT\s*\([^,]*,*\(([^)]+)\)\s*\)")
m = regex.search(text)
if m:
    print("Match:", m.group(1))
else:
    print("No match")

File: scratch/test_regex.js
const fs = require('fs');
const text = fs.readFileSync('test-data/Shapr3D/shapr3d_export_2026-06-11_11h15m.step', 'utf8');
const regex = /CARTESIAN_POINT\s*\([^,]*,\s*\(([^)]+)\)\s*\)/g;
let match;
let count = 0;
while ((match = regex.exec(text)) !== null) {
    count++;
}
console.log('Points found:', count);
